### ISSUE TITLE
Where is fl_choice?

### ISSUE THREAD

**Issue by ericu** [deech/fltkhs#131]
I expected to find the translation of fl_choice [https://www.fltk.org/doc-1.4/group__group__comdlg.html#ga36c19b29fcd5c261f3eafdf9f7126b16] in http://hackage.haskell.org/package/fltkhs-0.8.0.2/docs/Graphics-UI-FLTK-LowLevel-Ask.html.  Is it somewhere else?  I can't find it in any of the places I thought were likely.

**Comment by deech** [comment]
It is missing. Are you blocked on it? If you don't want to wait for me to do it, I can help you get a PR together.

**Comment by ericu** [comment]
I'm not blocked.  I have way too many features waiting to be implemented to get blocked that easily =*).  I'm just working on other stuff until that becomes available.

OTOH if it would be easy for you I'd be happy to take direction on a PR.

**Comment by deech** [comment]
I now remember why I didn't add `fl_choice`; it used [`varargs`](https://github.com/fltk/fltk/blob/master/src/fl_ask.cxx#L424) to create buttons and there's no way to bind to kind of function without having a bunch of `fl_choice2`, `fl_choice3`, ... functions. Since that was pain and I didn't see much need for it at the time I held off. Given this, do you still need this dialog?

**Comment by ericu** [comment]
My use case is the standard "You've made changes: Exit without saving/Save/Cancel" dialog.  While I could make a custom dialog for that, it seems a common enough case to have something like fl_ask implemented.  Likewise a two-button dialog is needed quite often.

I don't actually need the printf-style formatting, though.  It seems a bit weird to me to have that built into fl_ask when you could always just sprintf yourself before calling it.  A simplified version that took Text for the message and first button and Maybe Text for the other two buttons would be plenty for me.

**Comment by ericu** [comment]
Similarly, while I don't need printf-style formatting in fl_input, it would be nice to be able to supply the default string there.  I use it for renaming things, and if the default string were pre-filled, it would be nicer for making small edits.  Should I open a separate bug, or is this close enough to keep them together?

**Comment by deech** [comment]
A separate issue would be good. Thanks!

**Comment by ericu** [comment]
https://github.com/deech/fltkhs/pull/139
